__tests__/db.test.js

```diff
@@ -52,8 +106,219 @@ describe("GET /api", () =>{
+                expect(article).toHaveProperty("author")
+                expect(article).toHaveProperty("title")
+                expect(article).toHaveProperty("topic")
+                expect(article).toHaveProperty("created_at")
+                expect(article).toHaveProperty("votes")
+                expect(article).toHaveProperty("article_img_url")
+                expect(article).not.toHaveProperty('body')
+            })
+        })
+    })
+    test("Articles contain comment_count, a numeric count of comments referencing that ID within the comments table", () =>{
+        return request(app)
+        .get('/api/articles')
+        .then(({body})=>{
+            expect(body.articles.length > 0).toBe(true)
+            body.articles.forEach((article)=>{
+                expect(article).toHaveProperty("comment_count", expect.any(Number))
+            })
+        })
+    })
+    test("Articles return sorted by their date in decending order", () =>{
+        return request(app)
+        .get('/api/articles')   
+        .then(({body})=>{
+            expect(new Date(body.articles[0].created_at).getTime() >= new Date(body.articles[1].created_at).getTime()).toBe(true)
+            expect(new Date(body.articles[1].created_at).getTime() >= new Date(body.articles[2].created_at).getTime()).toBe(true)
+            expect(new Date(body.articles[2].created_at).getTime() >= new Date(body.articles[3].created_at).getTime()).toBe(true)
+            expect(new Date(body.articles[3].created_at).getTime() >= new Date(body.articles[4].created_at).getTime()).toBe(true)
+            expect(new Date(body.articles[4].created_at).getTime() >= new Date(body.articles[6].created_at).getTime()).toBe(true)
+        })
+    })
+    test("Bad Queries do not effect outcome length or status", () =>{
+        return request(app)
+        .get('/api/articles?article_id=ga')
+        .then((res)=>{
+                expect(res.statusCode).toBe(200)
+                expect(res.body.articles.length > 0)
+        })
+    })
+  })
+describe("GET /api/articles/", () => {
+    test("Calls appropriate function with 200 status code", () =>{
+        return request(app)
+        .get('/api/articles/1')
+        .then((res)=>{
+            expect(res.statusCode).toBe(200)
+        })
+    })
+    test("Returns an object in the body", () =>{
+        return request(app)
+        .get('/api/articles/1')
+        .then(({body})=>{
+            expect(typeof body).toBe("object")
+        })
+    })
+    test("Returns 400 status code when requesting a non-numeric ID", () =>{
+        return request(app)
+        .get('/api/articles/fish')
+        .then((res)=>{
+            expect(res.statusCode).toBe(400)
+        })
+    })
+    test("Returns 404 status code when requesting an article ID with no data", () =>{
+        return request(app)
+        .get('/api/articles/135477')
+        .then((res)=>{
+            expect(res.statusCode).toBe(404)
+        })
+    })
```

Comment:
This article technically doesn't exist - so I would have a go at changing the describe